### DISCUSSION TITLE
[Mia-Platform console][Template or Example] Dynamic .gitignore generation using APIs

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#207] [Ideas]
Hi guys,
I noticed that we do not have a standard for the `.gitignore` file, which is very important.

Why we don't integrate a step to dynamically generate the `.gitignore` using `gitignore.io` APIs?

E.g. creating a Node.JS example/template the Mia-Platform Console's backend service could call the following API:
[https://www.toptal.com/developers/gitignore/api/node,visualstudiocode,webstorm,yarn](https://www.toptal.com/developers/gitignore/api/node,visualstudiocode,webstorm,yarn).

P.S. This approach can be used for other files as the `.dockerignore`.

**Comment by davidebianchi** [comment]
Hi @silversoul93, this could be a good idea to improve the marketplace. But probably the template already have a gitignore file, what do you think?

**Reply by frengooo** [reply]
Hi @davidebianchi, thanks for your reply.
Yes, templates and examples usually already have the `.gitignore` file but very often something inside it is missing (e.g. IDE related folders like the `.idea` one, package manager related files and so on).

Moreover, I don't think someone is maintaining these files updated and, with the integration I am talking about, probably no one will need to do that.

**Reply by frengooo** [reply]
@davidebianchi did you lose my reply? 😂

**Comment by akelity** [comment]
This feature has been positively evaluated and may be considered for inclusion in the roadmap in the future, depending on priorities and overall specific needs.
